### DISCUSSION TITLE
Where to find/store pg_hba.conf?

### DISCUSSION THREAD

**Discussion by decheu** [QGEP/QGEP#714] [Q&A] [answered]
https://qgep.github.io/docs/installation-guide/server.html#server-installation says to change the pg_hba.conf
https://www.postgresql.org/docs/devel/auth-pg-hba-conf.html says it's `stored in the database cluster's data directory`
When I search my entire mac for pg_hba.conf, I can't find anything.
Or do I have to put it somewhere?

**Comment by sjib** [comment]
https://www.postgresql.org/docs/current/auth-pg-hba-conf.html says 
`A default pg_hba.conf file is installed when the data directory is initialized by initdb. It is possible to place the authentication configuration file elsewhere, however; see the hba_file configuration parameter.`

**Reply by decheu** [reply]
Ok, got the first part: pg_hba.conf must be somewhere. But when I run a search over my entire mac, finder doesn't find a pg_hba.conf. Could it be hidden?
https://www.postgresql.org/docs/current/runtime-config-file-locations.html#GUC-HBA-FILE taught me quite some things but didn't help me in finding pg_hba.conf.

**Comment by decheu** [comment]
Apparently, pg_hba.conf is optional and QGEP runs nicely without. --> see suggested change of https://qgep.github.io/docs/installation-guide/server.html#server-installation

**Comment by sjib** [comment, marked as answer]
Here some instructions for Mac OS
https://til.codes/postgresql-how-to-find-pg_hba-conf-file-using-mac-os-x/

**Comment by sjib** [comment]
And this shows the configuration when you are in pg_admin and can run a Query Window session:

pg_hba.conf : `SHOW hba_file;`

![show_hba_config_path](https://user-images.githubusercontent.com/1265865/152370537-438b0c8c-30fa-439f-a419-af66bf49cc47.jpg)


Same for postgressql.conf:  `SHOW config_file;`
![show_pg_config_path](https://user-images.githubusercontent.com/1265865/152370897-92e0ae8b-7b95-41e9-b9e1-e091e6633e66.jpg)
